File: frontend/src/types/category.ts
export type MajorCategory = {
  id: number;
  name: string;
  sort_order: number;
};

export type ExpenseCategory = {
  id: number;
  major_category_id: number;
  name: string;
  sort_order: number;
};

export type IncomeCategory = {
  id: number;
  name: string;
  sort_order: number;
};

File: backend/app/schemas/category.py
from pydantic import BaseModel, ConfigDict, Field


class MajorCategoryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    sort_order: int


class ExpenseCategoryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    major_category_id: int
    name: str
    sort_order: int


class ExpenseCategoryCreate(BaseModel):
    major_category_id: int
    name: str = Field(min_length=1, max_length=50)


class IncomeCategoryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    sort_order: int


class IncomeCategoryCreate(BaseModel):
    name: str = Field(min_length=1, max_length=50)
